hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/bucket/ConsistentBucketIndexUtils.java

```diff
@@ -184,12 +184,31 @@ public static boolean saveMetadata(HoodieTable table, HoodieConsistentHashingMet
     StoragePath dir = FSUtils.constructAbsolutePath(
         table.getMetaClient().getHashingMetadataPath(), metadata.getPartitionPath());
     StoragePath fullPath = new StoragePath(dir, metadata.getFilename());
+    byte[] bytes;
     try {
-      storage.createImmutableFileInPath(fullPath, Option.of(metadata.toBytes()));
-      return true;
+      bytes = metadata.toBytes();
     } catch (IOException e) {
+      LOG.error("Error when converting hashing metadata: {} to bytes", metadata.getPartitionPath(), e);
+      throw new HoodieIndexException("Error while converting hashing metadata to bytes", e);
+    }
+    try {
+      storage.createImmutableFileInPath(fullPath, Option.of(bytes));
+      return true;
+    } catch (HoodieIOException e) {
       LOG.warn("Failed to update bucket metadata: " + metadata, e);
-      return false;
+      boolean exist;
+      try {
+        exist = storage.exists(fullPath);
```

Comment:
still return success for creating failed but the target file already exists.

Reply:
can you elaborate why the test case fails without changing the parallelism?

Reply:
> can you elaborate why the test case fails without changing the parallelism?

For example, if the parallelism level is 2, two tasks load the index mapping relationship respectively. task1 finds that there is no valid metadata when loading metadata, and creates an initial metadata file. When task1 creates the file but has not finished writing the content, task2 loads the mapping and reads the empty file created by task1, causing an exception